packages/dx-grid-core/src/plugins/table-summary-row/helpers.js

```diff
@@ -14,7 +14,36 @@ export const isTotalSummaryTableRow = tableRow => tableRow.type === TABLE_TOTAL_
 export const isGroupSummaryTableRow = tableRow => tableRow.type === TABLE_GROUP_SUMMARY_TYPE;
 export const isTreeSummaryTableRow = tableRow => tableRow.type === TABLE_TREE_SUMMARY_TYPE;
 
-export const getColumnSummaries = (summaryItems, columnName, summaryValues) => summaryItems
-  .map((item, index) => [item, index])
-  .filter(([item]) => item.columnName === columnName)
-  .map(([item, index]) => ({ type: item.type, value: summaryValues[index] }));
+export const getColumnSummaries = (
+  summaryItems, columnName, summaryValues, predicate = () => true,
+) => (
+  summaryItems
+    .map((item, index) => [item, index])
+    .filter(([item]) => item.columnName === columnName && predicate(item))
+    .map(([item, index]) => ({ type: item.type, value: summaryValues[index] }))
+);
+
+const isInlineGroupSummary = summaryItem => (
+  summaryItem.showInGroupCaption || summaryItem.showInGroupRow
+);
+
+export const getGroupInlineSummaries = (summaryItems, tableColumns, summaryValues) => {
+  if (!summaryItems.some(isInlineGroupSummary)) {
+    return [];
+  }
+
+  return tableColumns.reduce((acc, col) => {
+    const colName = col.column.name;
+    const summaries = getColumnSummaries(
+      summaryItems, colName, summaryValues, item => item.showInGroupCaption,
```

Comment:
`item => item.showInGroupCaption` can be declared once (like `isInlineGroupSummary`)